### ISSUE TITLE
DoorKey boards don't match MiniGrid's: door row range and the -Random- goal

### ISSUE THREAD

**Issue by xweinp** [epignatelli/navix#225]
_Posted by an AI agent (Claude Opus 5, via Claude Code) on behalf of @xweinp._

## Summary

`Navix-DoorKey-Random-*` boards are not drawn the way MiniGrid's `DoorKeyEnv._gen_grid` draws them, and neither are the door rows of the plain `Navix-DoorKey-*` IDs. MiniGrid has no fixed-start DoorKey, so the plain IDs' agent at `(1, 1)` facing east is a NAVIX variant and not part of this issue. Everything else about the board has a MiniGrid reference.

For reference, MiniGrid (3.1.0) draws a board like this:

```python
self.put_obj(Goal(), width - 2, height - 2)
splitIdx = self._rand_int(2, width - 2)
self.grid.vert_wall(splitIdx, 0)
self.place_agent(size=(splitIdx, height))        # uniform cell left of the wall, uniform direction
doorIdx = self._rand_int(1, height - 2)
self.put_obj(Door("yellow", is_locked=True), splitIdx, doorIdx)
self.place_obj(obj=Key("yellow"), top=(0, 0), size=(splitIdx, height))
```

## Differences in `DoorKey._reset` (4.0.0)

1. **Door row, all IDs.** `door_row = jax.random.randint(k3, (), 1, self.height - 1)` can put the door in the bottom interior row, `height - 2`. MiniGrid's `doorIdx = _rand_int(1, height - 2)` never does, so its door row is one of `1..height-3`.
2. **Goal, `-Random-` IDs.** `goal_pos = random_positions(k2, second_room)` puts the goal on a random cell of the second room. MiniGrid always puts it in the bottom-right corner, `(height - 2, width - 2)`, which the plain IDs already do.
3. **One key for three draws, `-Random-` IDs.** `k2` is used for the player's direction, the goal position and the key position. MiniGrid draws them independently.

The wall column (`2..width-3`), the uniform player cell and direction on `-Random-` IDs, and the key position (uniform left of the wall, never on the player) already match.

## Not in scope

These concern the task rather than the board, as in #223. MiniGrid's `max_steps` is `10 * size**2`, where NAVIX's default is `4 * height * width`. MiniGrid's reward shrinks with the number of steps taken. MiniGrid does not use up the key when the door opens, and its doors can be closed again.
